File: Chapter03/Java/TestResources/app/src/main/java/com/ai/android/book/resources/MyApplication.java
package com.ai.android.book.resources;

import android.app.Application;
import android.content.res.Configuration;
import android.util.Log;

import androidx.annotation.NonNull;

public class MyApplication extends Application {

    private static final String TAG = MyApplication.class.getSimpleName();

    @Override
    public void onConfigurationChanged(@NonNull Configuration newConfig) {
        super.onConfigurationChanged(newConfig);
        Log.d(TAG, "configuration changed");
    }

    @Override
    public void onCreate() {
        super.onCreate();
        Log.d(TAG, "onCreate");
    }

    @Override
    public void onLowMemory() {
        super.onLowMemory();
        Log.d(TAG, "onLowMemory");
    }

    @Override
    public void onTerminate() {
        super.onTerminate();
        Log.d(TAG, "onTerminate");
    }
}

File: Chapter03/Kotlin/TestResources/app/src/main/java/com/ai/android/book/resources/MyApplication.kt
package com.ai.android.book.resources

import android.app.Application
import android.content.res.Configuration
import android.util.Log

class MyApplication: Application(){

    override fun onConfigurationChanged(newConfig: Configuration) {
        super.onConfigurationChanged(newConfig)
        Log.d(tag, "configuration changed")
    }

    override fun onCreate() {
        super.onCreate()
        Log.d(tag, "onCreate")
    }

    override fun onLowMemory() {
        super.onLowMemory()
        Log.d(tag, "onLowMemory")
    }

    override fun onTerminate() {
        super.onTerminate()
        Log.d(tag, "onTerminate")
    }

    companion object {
        const val tag = "MyAplication"
    }
}